### PR TITLE
[watarmark] add watermark support

### PR DESCRIPTION
**Pull request by mykolaf**
Add code for Watermark feature
[High Level Design Document](https://github.com/Azure/SONiC/blob/694754ac31c77724f43458e2779b552223791974/doc/watermarks_HLD.md)

*What I did*

- Extended Flex counter to collect PG stats
- Extended Flex counter to generate PG MAPS(PG to port, PG to index, PG to name)
- Read and clear stats mode support

### CODE REVIEW

**Comment by mykolaf** [comment]
Builds failed due to outdated schema. Dependency on sonic-swss-common.

**Comment by lguohan** [comment]
retest this please

File: syncd/syncd.cpp [hunk 1]
```diff
@@ -2857,6 +2873,29 @@ void processFlexCounterEvent(
...
+                std::vector<sai_ingress_priority_group_stat_t> pgCounterIds;
+                for (const auto &str : idStrings)
+                {
+                    sai_ingress_priority_group_stat_t stat;
+                    sai_deserialize_ingress_priority_group_stat(str, stat);
+                    pgCounterIds.push_back(stat);
+                }
+                FlexCounter::setPriorityGroupCounterList(vid, rid, groupName, pgCounterIds);
+            }
+            else if (objectType == SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP && field == PG_ATTR_ID_LIST)
```
Comment (wendani) [inline syncd/syncd.cpp]: Is PG_ATTR_ID_LIST field used by watermark?

Comment (mykolaf) [reply syncd/syncd.cpp]: It's not used by watermark, but the other objects types (queue, port) have both  SAI_STAT and SAI_ATTR handlers, so implemented it for priority-groups as well.

**Pull request merged**